### ISSUE TITLE
Building error in v1.24.0

### ISSUE THREAD

**Issue by MrEvilCookie** [hypersad/atom-armv7l#5]
I removed all files that I had from old version, and followed readme.md instructions, but it errored while installing apm:
pi@raspberrypi:~/atom $ script/build                  
Node:   v9.4.0                                        
Npm:    v5.3.0                                        
Installing script dependencies                        
Installing apm                                        
npm ERR! path /home/pi/atom/apm/node_modules/atom-pack
age-manager/node_modules/npm/node_modules/ansicolors  
npm ERR! code ENOENT                                  
npm ERR! errno -2                                     
npm ERR! syscall rename                               
npm ERR! enoent ENOENT: no such file or directory, ren
ame '/home/pi/atom/apm/node_modules/atom-package-manag
er/node_modules/npm/node_modules/ansicolors' -> '/home
/pi/atom/apm/node_modules/atom-package-manager/node_mo
dules/npm/node_modules/.ansicolors.DELETE'            
npm ERR! enoent This is related to npm not being able 
to find a file.                                       
npm ERR! enoent                                       
                                                      
npm ERR! A complete log of this run can be found in:  
npm ERR!     /home/pi/.npm/_logs/2018-02-24T08_27_30_3
66Z-debug.log                                         
child_process.js:595                                  
    throw err;                                        
    ^                                                 
                                                      
Error: Command failed: /home/pi/atom/script/node_modul
es/.bin/npm --global-style --loglevel=error install   
npm ERR! path /home/pi/atom/apm/node_modules/atom-pack
age-manager/node_modules/npm/node_modules/ansicolors  
npm ERR! code ENOENT                                  
npm ERR! errno -2                                     
npm ERR! syscall rename                               
npm ERR! enoent ENOENT: no such file or directory, ren
ame '/home/pi/atom/apm/node_modules/atom-package-manag
er/node_modules/npm/node_modules/ansicolors' -> '/home
/pi/atom/apm/node_modules/atom-package-manager/node_mo
dules/npm/node_modules/.ansicolors.DELETE'            
npm ERR! enoent This is related to npm not being able 
to find a file.                                       
npm ERR! enoent                                       
                                                      
npm ERR! A complete log of this run can be found in:  
npm ERR!     /home/pi/.npm/_logs/2018-02-24T08_27_30_3
66Z-debug.log                                         
                                                      
    at checkExecSyncError (child_process.js:574:11)   
    at Object.execFileSync (child_process.js:592:13)  
    at module.exports (/home/pi/atom/script/lib/instal
l-apm.js:9:16)                                        
    at Object.<anonymous> (/home/pi/atom/script/bootst
rap:29:1)                                             
    at Module._compile (module.js:660:30)             
    at Object.Module._extensions..js (module.js:671:10
)                                                     
    at Module.load (module.js:573:32)                 
    at tryModuleLoad (module.js:513:12)               
    at Function.Module._load (module.js:505:3)        
    at Module.require (module.js:604:17)              
pi@raspberrypi:~/atom $

**Comment by MrEvilCookie** [comment]
Nevermind, removed package-lock.json and it worked.

**Comment by MrEvilCookie** [comment]
Now it fails here:
Installing grammar-selector@0.49.8 ✗             
                                                 
npm ERR! tar.unpack error reading /tmp/d-118124-2
3565-by2x0f.rgquvm42t9/package.tgz               
npm ERR! addLocal Could not install /tmp/d-118124
-23565-by2x0f.rgquvm42t9/package.tgz             
npm ERR! Linux 4.9.35-v7+                        
npm ERR! argv "/home/pi/atom/apm/node_modules/ato
m-package-manager/bin/node" "/home/pi/atom/apm/no
de_modules/atom-package-manager/node_modules/npm/
bin/npm-cli.js" "--globalconfig" "/home/pi/.atom/
.apm/.apmrc" "--userconfig" "/home/pi/.atom/.apmr
c" "install" "/tmp/d-118124-23565-by2x0f.rgquvm42
t9/package.tgz" "--runtime=electron" "--target=1.
6.16" "--arch=arm"                               
npm ERR! node v6.9.5                             
npm ERR! npm  v3.10.10                           
                                                 
npm ERR! 0-byte tarball                          
npm ERR! Please run `npm cache clean`            
npm ERR!                                         
npm ERR! If you need help, you may report this er
ror at:                                          
npm ERR!     <https://github.com/npm/npm/issues> 
                                                 
npm ERR! Please include the following file with a
ny support request:                              
npm ERR!     /home/pi/atom/npm-debug.log         
                                                 
child_process.js:595                             
    throw err;                                   
    ^                                            
                                                 
Error: Command failed: /home/pi/atom/apm/node_mod
ules/atom-package-manager/bin/apm --loglevel=erro
r install                                        
    at checkExecSyncError (child_process.js:574:1
1)                                               
    at Object.execFileSync (child_process.js:592:
13)                                              
    at module.exports (/home/pi/atom/script/lib/r
un-apm-install.js:14:16)                         
    at Object.<anonymous> (/home/pi/atom/script/b
ootstrap:35:1)                                   
    at Module._compile (module.js:660:30)        
    at Object.Module._extensions..js (module.js:6
71:10)                                           
    at Module.load (module.js:573:32)            
    at tryModuleLoad (module.js:513:12)          
    at Function.Module._load (module.js:505:3)   
    at Module.require (module.js:604:17)         
pi@raspberrypi:~/atom $

**Comment by hypersad** [comment]
Yep, first problem caused by 'apm/package-lock.json` and can be solved by removing it. There is a new script that should remove unused lockfiles, but I'm not sure when it should execute, so I recommend you to delete it manually before building if it exists.

To solve second problem try to execute `atom/clean` script, and I highly recommend to upgrade Node.JS and NPM.

**Comment by MrEvilCookie** [comment]
I updated node to 9.6.1 and npm to 5.6.0, executed script/clean and removed apm/package-lock.js. Everything looks fine right now

**Comment by MrEvilCookie** [comment]
This one is long: https://paste.ee/p/CKvUk

**Comment by hypersad** [comment]
I have no idea how to deal with it :(

You could use prebuilt `.deb` package or archive from hypersad/atom-armv7l#4.

**Comment by MrEvilCookie** [comment]
I tried prebuilt, I don't quite remember what error was that, but I'ІІ try recreating that

**Comment by apenaz** [comment]
prebuild-install WARN install No prebuilt binaries found (target=6.9.5 runtime=node arch=arm platform=linux)
apm  1.18.12
npm  3.10.10
node 6.9.5 arm
atom unknown
python 2.7.13
git 2.11.0
Installing modules ✗
> atom@1.24.0 preinstall /home/pi/atom
> node -e 'process.exit(0)'


> keyboard-layout@2.0.13 install /home/pi/atom/node_modules/keyboard-layout
> node-gyp rebuild


npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@2.0.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated coffee-script@1.8.0: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.4.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated formatio@1.1.1: This package is unmaintained. Use @sinonjs/formatio instead
npm WARN prefer global typescript@1.4.1 should be installed with -g
npm WARN prefer global jasmine-node@1.10.2 should be installed with -g
child_process.js:596
    throw err;
    ^

Error: Command failed: /home/pi/atom/apm/node_modules/atom-package-manager/bin/apm --loglevel=error install
    at checkExecSyncError (child_process.js:575:11)
    at Object.execFileSync (child_process.js:593:13)
    at module.exports (/home/pi/atom/script/lib/run-apm-install.js:14:16)
    at Object.<anonymous> (/home/pi/atom/script/bootstrap:35:1)
    at Module._compile (module.js:662:30)
    at Object.Module._extensions..js (module.js:673:10)
    at Module.load (module.js:575:32)
    at tryModuleLoad (module.js:515:12)
    at Function.Module._load (module.js:507:3)
    at Module.require (module.js:606:17)

**Comment by hypersad** [comment]
@apenaz I will start building on my Raspberry Pi 3 Model B now and check, if it really fails or it fails only in your case. Quick question: did you try to remove `apm/package-lock.json` and start building again?
